src/components/SortDropdown.js

```diff
@@ -0,0 +1,157 @@
+    width: '32px',
+    height: '29px',
+  },
+  iconGridStyle: {
+    margin: 'auto',
+    paddingTop: '6px',
+  },
+  optionContainerStyle: {
+    display: 'flex',
+    minHeight: '43px',
+    minWidth: '177px',
+    backgroundColor: theme.palette.background.default,
+    border: '0.5px solid #BCBCBC',
+    cursor: 'pointer',
+    '&:hover': {
+      backgroundColor: theme.palette.secondary.light,
+    },
+  },
+  optionsGridStyle: {
+    position: 'absolute',
+    borderRadius: '0px 0px 4px 4px',
+    left: '0',
+    right: '0',
+  },
+  dropdownContainer: {
+    position: 'relative',
+    display: 'inline-block',
+  },
+}));
+
+export const SortDropdown = ({
+  inputSortMethodList,
+  defaultSortMethod,
+  setSortMethod,
+}) => {
+  const classes = useStyles();
+  const [isDropdownExpand, setIsDropdownExpand] = useState(false);
+  const [selectedSortMethod, setSelectedSortMethod] =
+    useState(defaultSortMethod);
+  const sortMethodMap = {
+    'best match': 'Best Match',
+    'updated': 'Last Updated',
+    'stars': 'Stargazer Count',
+  };
+
+  const handleOptionChange = (event) => {
+    const sortMethodVal = event.target.dataset.value;
+    setSelectedSortMethod(sortMethodVal);
+    setSortMethod(sortMethodVal);
+    setIsDropdownExpand(!isDropdownExpand);
+  };
+
+  const toggleDropdown = () =>{
+    setIsDropdownExpand(!isDropdownExpand);
+  }
+
+  const generateOptions = () => {
+    return inputSortMethodList.map((sortMethod) => {
+      let isSortMethodSelected = false;
+      const optionName = sortMethodMap[sortMethod];
+      if (selectedSortMethod === sortMethod) {
+        isSortMethodSelected = true;
+      }
+      return (
+        <Grid
+          item
+          key={sortMethod}
+          data-value={sortMethod}
+          onClick={handleOptionChange}
+          className={
+            isSortMethodSelected
+              ? classes.hiddenStyle
+              : classes.optionContainerStyle
+          }
+        >
+          <Typography
+            onClick={handleOptionChange}
+            data-value={sortMethod}
+            className={
+              isSortMethodSelected
+                ? classes.hiddenStyle
+                : classes.optionTextStyle
+            }
+          >
+            {optionName}
+          </Typography>
+        </Grid>
+      );
+    });
+  };
+
+  return (
+    <Box className={classes.dropdownContainer}>
+      <Grid item data-cy='sort-dropdown' className={classes.dropdown} onClick={toggleDropdown}>
+        <Grid className={classes.selectedOptionGridStyle}>
+          <Typography className={classes.selectedOptionTextStyle}>
+            {'Sort: ' + sortMethodMap[selectedSortMethod]}
+          </Typography>
+        </Grid>
+        <Grid className={classes.iconGridStyle}>
+          {isDropdownExpand ? (
+            <ExpandLessIcon className={classes.dropdownIconStyle} />
+          ) : (
+            <ExpandMoreIcon className={classes.dropdownIconStyle} />
+          )}
```

Comment:
You can try putting this conditional rendering inside of a function, and then calling the function along with passing in the isDropdownExpand variable. This should help bring down the Cognitive Complexity error that CodeClimate is throwing.
```
const renderExpandIcon = (isDropdownExpand) => {
  if (isDropdownExpand) {
    return <ExpandLessIcon className={classes.dropdownIconStyle} />
  } else {
    return <ExpandMoreIcon className={classes.dropdownIconStyle} />
  }
}
```
^ Insert that function outside of SortDropdown (into global scope) and then call it as shown below

```suggestion
        <Grid className={classes.iconGridStyle}>
          {renderExpandIcon(isDropdownExpand)}
```